stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -218,6 +221,36 @@ public static <ReqT, RespT> Iterator<RespT> blockingServerStreamingCall(
     return result;
   }
 
+  /**
+   * Initiate a bidirectional-streaming {@link ClientCall} and returning a stream object
+   * ({@link BlockingClientCall}) which can be used by the client to send and receive messages over
+   * the grpc channel.
+   *
+   * @return an object representing the call which can be used to read, write and terminate it.
+   */
+  public static <ReqT, RespT> BlockingClientCall<ReqT, RespT> blockingBidiStreamingCall(
+      Channel channel, MethodDescriptor<ReqT, RespT> method, CallOptions callOptions) {
+    ThreadlessExecutor executor = new ThreadlessExecutor();
+    ClientCall<ReqT, RespT> call = channel.newCall(method,
+        callOptions.withOption(ClientCalls.STUB_TYPE_OPTION, StubType.BLOCKING)
```

Comment:
Because users can go around the stubs and directly to ClientCalls, we can't remove.  Is there a reason why when the option already exists with the same value we don't just return "this" from withOption rather than building an entirely new instance?